src/app/extensions/rating/shared/product-review-create-dialog/product-review-create-dialog.component.ts

```diff
@@ -0,0 +1,138 @@
+import { ChangeDetectionStrategy, Component, OnInit, TemplateRef, ViewChild } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import { NgbModal, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
+import { FormlyFieldConfig } from '@ngx-formly/core';
+import { Observable, map, tap } from 'rxjs';
+
+import { AccountFacade } from 'ish-core/facades/account.facade';
+import { markAsDirtyRecursive } from 'ish-shared/forms/utils/form-utils';
+
+import { ProductReviewsFacade } from '../../facades/product-reviews.facade';
+import { ProductReview } from '../../models/product-reviews/product-review.model';
+
+@Component({
+  selector: 'ish-product-review-create-dialog',
+  templateUrl: './product-review-create-dialog.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ProductReviewCreateDialogComponent implements OnInit {
+  /**
+   *  A reference to the current modal.
+   */
+  modal: NgbModalRef;
+
+  @ViewChild('modal') modalTemplate: TemplateRef<unknown>;
+
+  form = new FormGroup({});
+  submitted = false;
+  fields: FormlyFieldConfig[];
+  model$: Observable<Partial<ProductReview>>;
+
+  constructor(
+    private ngbModal: NgbModal,
+    private accountFacade: AccountFacade,
+    private reviewFacade: ProductReviewsFacade
+  ) {}
+
+  ngOnInit() {
+    this.model$ = this.accountFacade.user$.pipe(
+      tap(user => {
```

Comment:
Can you elaborate on which edge case this prevents?

Reply:
If the anonymous user clicks 'Rate this product' link the login dialog is shown. If he logs in it is necessary to get the reviews again, only this way it is possible to show him his own review separately.